quodlibet/util/collection.py

```diff
@@ -80,8 +135,15 @@ def bayesian_average(nums, c=None, m=None):
     "max": max,
     "min": min,
     "sum": sum,
-    "avg": avg,
-    "bav": bayesian_average
+    "avg": weighted_average,
+    "bav": weighted_bayesian_average,
+    "unweighted_avg": unweighted_average,
```

Comment:
I was thinking `uav` (or `uavg`) - seems more consistent?

Reply:
It's true that a long suffix is uncommon (all other are 3), and so maybe less consistent, but I feel like in this case it's warranted. If we only have `uavg`, `avg, `ubav` and `bav`, users would have to look up the meaning (or just ignore them). With `unweighted` it's immediately clear that the other method with the same name is weighted (though not by what). Also, the longer names means users are more likely to use `avg`, which in most cases is what they want anyway.

Reply:
I see. `bav` might want to be `bayesian_avg` with this thinking then I guess..

Reply:
I'd agree with that, though then we have to go for `unweighted_bayesian_avg` :smile:  which is too long. I get your point though.. It's a bit arbitrary what's abbreviated. I'd actually have gone with `bavg`, but by changing that we'd break existing queries if we switched to that long version.

Reply:
at the time I was thinking 3-letters is the main thing I guess

Reply:
Actually remind me why we're adding an `unweighted_` modifier and not just a `weighted_` version (thus leaving current functionality / queries as-is)...

Thus `avg` (remains same) and `bav` (remains same) and `wav` (or `weighted_avg` or whatever)

Reply:
Main reason: Provides a reasonable update to ratings (and can easily be disabled) . At the same time, ratings don't change that much.

Also, and I may have misread the thread's sentiment, but it definitely seemed like most people were in favor of making this a default. See, for example, lazka's comment: https://github.com/quodlibet/quodlibet/issues/3048#issuecomment-438558504

https://github.com/quodlibet/quodlibet/blob/bc03a5a6393ccef53b280852051f3ffee5127594/quodlibet/util/collection.py#L189-L194

(minor reason: more consistent with bitrate)

Of course, there was the caveat that you mentioned (about long songs dominating the average), which made me think of the whole smoothing thing :slightly_smiling_face: